pulser/simulation/simulation.py

```diff
@@ -211,23 +211,44 @@ def _build_basis_and_op_matrices(self) -> None:
                 self.basis[proj[0]] * self.basis[proj[1]].dag()
             )
 
-    def _build_operator(self, operations: dict,
-                        ) -> qutip.Qobj:
-        """Create qutip.Qobj with nontrivial action at *qubit_ids."""
+    def build_operator(self, operations: list) -> qutip.Qobj:
+        """Creates an operator with non trivial actions on some qubits.
+        Takes as argument a list of tuples [(operator_1, qubits_1),
+        (operator_2, qubits_2)...]. Returns the operator given by
+        sum_i {operator_i applied on qubits_i and Id on the rest}.
+        (operator, 'global') returns the sum for all $j$ of operator
+        applied at qubit $j$ and identity elsewhere.
+
+        Example for 3 qubits:
+        - [(Z, [1, 2]), (Z, [2, 3])] returns ZZI + IZZ
+        - [(X, 'global')] returns XII + IXI + IIX
+
+        Args:
+            operations (list): List of tuples (operator, qubits)
+            operator can be a qutip.Quobj or a string key for self.op_matrix
+            qubits is the list on which operator will be applied. The qubits
+            can be passed as their index or their label in the register.
+
+        Returns:
+            qutip.Qobj: the final operator.
+        """
         op_list = [self.op_matrix['I'] for j in range(self._size)]
-        for operator, qubits in operations.items():
+
+        for operator, qubits in operations:
             if qubits == 'global':
-                return sum(self._build_operator({operator: [q_id]})
+                return sum(self.build_operator([(operator, [q_id])])
                            for q_id in self._qdict)
             else:
                 if len(set(qubits)) < len(qubits):
                     raise ValueError("Duplicate atom ids in argument list.")
+                if isinstance(operator, str):
+                    operator = self.op_matrix[operator]
                 for qubit in qubits:
```

Comment:
Check if `qubit` is either a valid `q_id` or a valid index (i.e. less than the number of atoms)